spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/convert/MappingMongoConverterTests.java

```diff
@@ -478,4 +481,130 @@ public String toString() {
+		}
+
+		@Test // GH-5065
+		@DisplayName("Test control: Converter should read an explicitly assigned null as a null map.")
+		void readsExplicitlyNullMapCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":null}");
+			//EmptyMapDocument target = converter.read(EmptyMapDocument.class, document);
+			assertThat(converter.read(EmptyMapDocument.class, document).map).isNull();
+			assertThat(converter.read(EmptyMapDocument.class, document).getMap()).isNull();
+		}
+
+		@Test // GH-5065
+		@DisplayName("Converter should read an empty object as an empty map when using @DocumentReference.")
+		void readsEmptyMapWithDocumentReferenceCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":{}}");
+			//DocumentReferenceEmptyMapDocument target = converter.read(DocumentReferenceEmptyMapDocument.class, document);
+			assertThat(converter.read(DocumentReferenceEmptyMapDocument.class, document).map).isNotNull().isEmpty();
+			assertThat(converter.read(DocumentReferenceEmptyMapDocument.class, document).getMap()).isNotNull().isEmpty();
+		}
+
+		@Test // GH-5065
+		@DisplayName("Converter should read an explicitly assigned null as a null map when using @DocumentReference.")
+		void readsExplicitlyNullMapWithDocumentReferenceCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":null}");
+			//DocumentReferenceEmptyMapDocument target = converter.read(DocumentReferenceEmptyMapDocument.class, document);
+			assertThat(converter.read(DocumentReferenceEmptyMapDocument.class, document).map).isNull();
+			assertThat(converter.read(DocumentReferenceEmptyMapDocument.class, document).getMap()).isNull();
+		}
+
+		@Test // GH-5065
+		@DisplayName("Converter should read an empty object as an empty map with a valid values property when using @DocumentReference(lazy = true).")
+		void readsEmptyMapWithLazyLoadedDocumentReferenceCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":{}}");
+			//LazyDocumentReferenceEmptyMapDocument target = converter.read(LazyDocumentReferenceEmptyMapDocument.class, document);
+			assertThat(converter.read(LazyDocumentReferenceEmptyMapDocument.class, document).map).isNotNull().isEmpty();
+			assertThat(converter.read(LazyDocumentReferenceEmptyMapDocument.class, document).getMap()).isNotNull().isEmpty();
+			assertThat(converter.read(LazyDocumentReferenceEmptyMapDocument.class, document).map.values()).isNotNull().isEmpty();
+			assertThat(converter.read(LazyDocumentReferenceEmptyMapDocument.class, document).getMap().values()).isNotNull().isEmpty();
+		}
+
+		@Test // GH-5065
+		@DisplayName("Converter should read an empty object as an empty map when using @DBRef.")
+		void readsEmptyMapWithDBRefCorrectly() {
+			org.bson.Document document = org.bson.Document.parse("{\"map\":{}}");
+			//DBRefEmptyMapDocument target = converter.read(DBRefEmptyMapDocument.class, document);
```

Comment:
What's up with the commented code line, and the repetition of that expression in every assert?

Why not introduce a `result` variable and assert on that?

Reply:
I wanted to ensure that the first `assertThat` call didn't have a side effect (such as lazy loading the map).  I can break this up into 2 tests.